src/cluster/src/check.rs

```diff
@@ -30,6 +31,51 @@ const RESOURCE_SERVICE_ACCOUNT: &str = "secret";
 /// The type of error that can occur while running preinstall checks
 #[derive(Error, Debug)]
 pub enum CheckError {
+    /// There was a problem with the helm client during pre-check
+    #[error("Helm client error")]
+    HelmError(#[from] HelmError),
+
+    /// There was a problem fetching kubernetes configuration
```

Comment:
Kubernetes errors already have their own type, it's the `K8ConfigError` you see here. The reason it is part of the `CheckError` variants is because it may be an underlying cause of a check failure, so it gets bundled up so that it may be inspected and debugged later.

I'm not sure what the call-to-action is here, I didn't actually add this error type in this PR, I only reformatted it. If there is something we would like to do about this, I think it would be good to make a new issue.